src/utils/generator.js

```diff
@@ -311,18 +311,19 @@ export class Generator {
               invariant(typeString !== undefined, typeValue);
               return t.binaryExpression(
                 "!==",
-                t.unaryExpression("typeof", accessedPropertyOf(objectNode), true),
+                t.unaryExpression("typeof", valueNode, true),
                 t.stringLiteral(typeString)
               );
             })
           );
           return checks.reduce((expr, newCondition) => t.logicalExpression("&&", expr, newCondition));
         };
+        this.emitInvariant([value, value], condition, valueNode => valueNode);
       }
     } else {
       condition = ([objectNode, valueNode]) => t.binaryExpression("!==", accessedPropertyOf(objectNode), valueNode);
+      this.emitInvariant([object, value, object], condition, objnode => accessedPropertyOf(objnode));
```

Comment:
Was there a particular reason to list `object` twice, maybe for referencing counting reasons?